### ISSUE TITLE
CD for production

### ISSUE THREAD

**Issue by shmish111** [input-output-hk/plutus#1278]
We want to be able to deploy to production easily. For now we can accept that we cannot make infrastructure changes automatically however we can make nixops changes. What we need is a `prod` branch which we create PRs against from `master`. When this is merged we trigger the CD server in the production environment. We will need to protect the `prod` branch so that only the `deployment-team` can merge into it.

**Comment by shmish111** [comment]
requires https://github.com/input-output-hk/plutus/issues/1277 and https://github.com/input-output-hk/plutus/issues/1276

**Comment by shmish111** [comment]
requires #1283 

Once all these issues are closed we will need to deploy `deploymentServer` to the prod nixops machine and set the `ref` value to `prod`. We will then need to create a `prod` branch that is protected.

When we want to deploy something we create a PR with `prod` as the target and upon merging `prod` branch will be deployed.

**Comment by shmish111** [comment]
Everything has now been done and is working on alpha environment, I have arranged for a production deployment tomorrow so after that we should be able to deploy to production by creating PRs that go into a special `prod` branch.

We will need someone with suitable plutus github repo permissions to protect the `prod` branch so that only certain people can merge PRs into it and so nobody can push to it. I was going to ask @michaelpj  to do that but he is away at the moment, I'm not sure who else has permission, perhaps @j-mueller or @mchakravarty ?

**Comment by michaelpj** [comment]
`prod` created based on 01986aa08c0d35ad38fa61edfeea891f8986135a, branch protected so only `plutus-admin` can push to it for now, we can create a `plutus-deployers` team if we want later.

**Comment by shmish111** [comment]
This has now been deployed but I would rather not close this issue until it's been tested at least once.

**Comment by michaelpj** [comment]
Did we actually try this in the end?

**Comment by shmish111** [comment]
This is still waiting on some fixes to the hacked together production terraform stuff. Deployments have been rare so the pain has gone unnoticed.
